transport/tchannel/transport.go

```diff
@@ -70,17 +78,19 @@ func NewTransport(opts ...TransportOption) (*Transport, error) {
 		return nil, fmt.Errorf("NewTransport does not accept WithChannel, use NewChannelTransport")
 	}
 
-	return options.newTransport(), nil
+	return options.newTransport()
 }
 
-func (o transportOptions) newTransport() *Transport {
+func (o transportOptions) newTransport() (*Transport, error) {
```

Comment:
The reason came and went, left this as a gift to commemorate its passing. None remember who or why.